test/transaction.ts

```diff
@@ -56,8 +65,8 @@ class FakeDatastoreRequest {
 
```

Comment:
@JustinBeckwith running into a slight bug that could use your quick input - on above lines [54] and [61] and below on line [191], we reassign the `save`, `delete`, and `rollback` request.ts methods to signatures that are being rejected for incompatibility

Basically, `delete` and `save` can't be reassigned to a signature that returns `void` because both are supposed to return promises when no callback is passed 
(Save: https://github.com/googleapis/nodejs-datastore/blob/master/src/request.ts#L962  
Delete: https://github.com/googleapis/nodejs-datastore/blob/master/src/request.ts#L342)

I can hack my way through and make the overloads return `void` instead of a promise, but that wouldn't be correct - was wondering what you'd recommend?

Reply:
Gah - yeah that can suck with the signature bits. I am ok with whatever casting you need in that situation.  I *do* like using sinon for the stubbing though, because I like having the option to check the `stub.calledOnce` stuff if I need to, and the security of knowing `sandbox.restore()` will undo all my shenanigans.

Reply:
I managed a casting approach, but I'm trying to make use of sinon for you (which means I have to learn Sinon again) - wanted to explain the delay